File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/ai/AiChatRequest.java
package fpt.aptech.server.dto.ai;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Builder;

/**
 * [1] AiChatRequest — DTO nhận yêu cầu chat từ ứng dụng Flutter.
 * <p>
 * Nhận nội dung văn bản (hoặc lời nói đã được chuyển thành văn bản).
 */
@Builder
public record AiChatRequest(
        // Bước 1: Validate tin nhắn không rỗng và độ dài tối đa
        @NotBlank(message = "Message cannot be empty")
        @Size(max = 2000, message = "Message must not exceed 2000 characters")
        @JsonProperty("message")
        String message,             // Nội dung tin nhắn của người dùng

        // Bước 2: Nhận loại đính kèm để phân biệt text/image/voice
        @JsonProperty("attachmentType")
        Integer attachmentType      // null = text | 1 = image | 2 = voice
) {
    @JsonCreator
    public static AiChatRequest create(
            @JsonProperty("message") String message,
            @JsonProperty("attachmentType") Integer attachmentType) {
        return new AiChatRequest(message, attachmentType);
    }
}

File: APP MOBILE/smart_money/lib/modules/ai/models/request/ai_chat_request.dart
// modules/ai/models/request/ai_chat_request.dart
// DTO nhận yêu cầu chat từ Flutter gửi lên backend
// Map đúng với AiChatRequest.java của Spring Boot

class AiChatRequest {
  final String message; // Nội dung tin nhắn của người dùng
  final int? attachmentType; // null = text | 1 = image | 2 = voice

  const AiChatRequest({
    required this.message,
    this.attachmentType,
  });

  /// Chuyển sang JSON để gửi lên server
  Map<String, dynamic> toJson() {
    return {
      'message': message,
      if (attachmentType != null) 'attachmentType': attachmentType,
    };
  }
}
